src/resolver.rs

```diff
@@ -4,14 +4,14 @@ use serde_json::Value;
 
 use crate::context::Context;
 use crate::error::ValidationError;
-use crate::schemas;
+use crate::schemas::{self, Draft};
 // TODO: Make the choice of resolver dynamic
 
 const DOCUMENT_PROTOCOL: &str = "document:///";
 
-fn id_of(draft_number: u8, schema: &Value) -> Option<&str> {
+fn id_of(draft: Draft, schema: &Value) -> Option<&str> {
     if let Value::Object(object) = schema {
-        if draft_number == 4 {
+        if draft == Draft::Draft4 {
```

Comment:
Just curious, really: Any difference between this and a `match` with a default branch at the end?

Reply:
Nearly none. For `==` we need the `PartialEq` trait implementation, which we wouldn't need to have for a match.
It will (most likely) compile down to the same thing.